modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/TypeScriptAngular2ClientCodegen.java

```diff
@@ -55,6 +57,74 @@ protected void addAdditionPropertiesToCodeGenModel(CodegenModel codegenModel, Mo
         addImport(codegenModel, codegenModel.additionalPropertiesType);
     }
 
+
+    /**
+     * Override of superclass method used to create a `vendorExtensions` entry
+     * on each model property for the code snippet that is used in the
+     * XyzModelSerializer.ts classes to deserialize the value.
+     *
+     * For example, this is used to deserialize date strings coming from a server
+     * into real JavaScript Date objects using the `new Date(dateStr)` constructor
+     * in the `serializer.mustache` template.
+     *
+     * @param objs
+     */
+    @Override
+    public Map<String, Object> postProcessAllModels(Map<String, Object> objs) {
+        objs = super.postProcessAllModels(objs);
+
+        List<CodegenModel> models = extractModels(objs);
+        for (CodegenModel model : models) {
+            for (CodegenProperty prop : model.vars) {
+                String snippet = createDeserializeSnippet(prop);
+
+                prop.vendorExtensions.put("deserializeSnippet", snippet);
```

Comment:
Is `vendorExtensions` the best way to add custom properties onto `CodegenProperty` objects, or is there something else I could use?

Context: This is used by the `serializer.mustache` template to inject code snippets for deserializing specific data types for response model properties. An example of this for the `Date` type is: `new Date(responseObj.theDate)`